pulp_ansible/tests/functional/api/collection/v3/test_sync.py

```diff
@@ -24,32 +24,54 @@ class SyncCollectionsFromPulpServerTestCase(TestCaseUsingBindings, SyncHelpersMi
 
     """
 
-    def test_sync_collections_from_pulp(self):
-        """Test sync collections from pulp server."""
+    def setUp(self):
+        """Set up the Sync tests."""
+        self.requirements_file = "collections:\n  - testing.k8s_demo_collection"
         body = gen_ansible_remote(
             url="https://galaxy.ansible.com",
-            requirements_file="collections:\n  - testing.k8s_demo_collection",
+            requirements_file=self.requirements_file,
         )
-        remote = self.remote_collection_api.create(body)
-        self.addCleanup(self.remote_collection_api.delete, remote.pulp_href)
+        self.remote = self.remote_collection_api.create(body)
+        self.addCleanup(self.remote_collection_api.delete, self.remote.pulp_href)
 
-        first_repo = self._create_repo_and_sync_with_remote(remote)
-        distribution = self._create_distribution_from_repo(first_repo)
+        self.first_repo = self._create_repo_and_sync_with_remote(self.remote)
+        self.distribution = self._create_distribution_from_repo(self.first_repo)
 
+    def test_sync_collections_from_pulp(self):
+        """Test sync collections from pulp server."""
         second_body = gen_ansible_remote(
-            url=distribution.client_url,
-            requirements_file="collections:\n  - testing.k8s_demo_collection",
+            url=self.distribution.client_url,
+            requirements_file=self.requirements_file,
         )
         second_remote = self.remote_collection_api.create(second_body)
         self.addCleanup(self.remote_collection_api.delete, second_remote.pulp_href)
 
         second_repo = self._create_repo_and_sync_with_remote(second_remote)
 
-        first_content = self.cv_api.list(repository_version=f"{first_repo.pulp_href}versions/1/")
+        first_content = self.cv_api.list(
+            repository_version=f"{self.first_repo.pulp_href}versions/1/"
+        )
         self.assertGreaterEqual(len(first_content.results), 1)
         second_content = self.cv_api.list(repository_version=f"{second_repo.pulp_href}versions/1/")
         self.assertGreaterEqual(len(second_content.results), 1)
 
+    def test_noop_resync_collections_from_pulp(self):
+        """Test sync collections from pulp server."""
+        second_body = gen_ansible_remote(
+            url=self.distribution.client_url,
+            requirements_file=self.requirements_file,
+        )
+        second_remote = self.remote_collection_api.create(second_body)
+        self.addCleanup(self.remote_collection_api.delete, second_remote.pulp_href)
+
+        second_repo = self._create_repo_and_sync_with_remote(second_remote)
+
+        # Resync
+        self._sync_repo(second_repo, remote=second_remote.pulp_href, optimize=True)
+
```

Comment:
I don't know how to assert the log, the test run on client side and the log happens on server side